spec/vcloud_director/generators/compute/instantiate_vapp_template_params_spec.rb

```diff
@@ -256,4 +256,134 @@ def self.common_cpu_assertions(xml)
       xml.xpath('./rasd:AllocationUnits').text.must_equal('hertz * 10^6')
     end
   end
+
+  describe 'vApp Network customization' do
+    [
+      {
+        :case          => 'minimal vapp networking',
+        :vapp_networks => [
+          {
+            :name   => 'network',
+            :subnet => [
+              {
+                :gateway => '1.2.3.4',
+                :netmask => '255.255.255.0'
+              }
+            ]
+          }
+        ],
+        :expect        => lambda do |vapp_net_conf|
+          common_vapp_network_config_assertions(vapp_net_conf)
+          vapp_net_conf.xpath('./vcloud:NetworkConfig/@networkName').text.must_equal 'network'
+          conf = vapp_net_conf.xpath('./vcloud:NetworkConfig/vcloud:Configuration')
+          conf.xpath('./vcloud:FenceMode').text.must_equal 'isolated'
+          conf.xpath('./vcloud:ParentNetwork').text.must_be_empty
+          subnet = conf.xpath('./vcloud:IpScopes/vcloud:IpScope')
+          subnet.xpath('./vcloud:Gateway').text.must_equal '1.2.3.4'
+          subnet.xpath('./vcloud:Netmask').text.must_equal '255.255.255.0'
+        end
+      },
+      {
+        :case          => 'full networking',
+        :vapp_networks => [
+          {
+            :name        => 'network',
+            :description => 'VM Network Description',
+            :deployed    => true,
+            :parent      => 'parent-network-id',
+            :fence_mode  => 'bridged',
+            :retain      => false,
+            :external_ip => '17.17.17.17',
+            :subnet      => [
+              {
+                :enabled    => true,
+                :inherited  => false,
+                :gateway    => '1.2.3.4',
+                :netmask    => '255.255.255.0',
+                :dns1       => '5.6.7.8',
+                :dns2       => '9.10.11.12',
+                :dns_suffix => 'dns-suffix',
+                :ip_range   => [{ :start => '192.168.254.100', :end => '192.168.254.199' }]
+              }
+            ]
+          }
+        ],
+        :expect        => lambda do |vapp_net_conf|
+          common_vapp_network_config_assertions(vapp_net_conf)
+          vapp_net_conf.xpath('./vcloud:NetworkConfig/@networkName').text.must_equal 'network'
+          vapp_net_conf.xpath('./vcloud:NetworkConfig/vcloud:Description').text.must_equal 'VM Network Description'
+          vapp_net_conf.xpath('./vcloud:NetworkConfig/vcloud:IsDeployed').text.must_equal 'true'
+          conf = vapp_net_conf.xpath('./vcloud:NetworkConfig/vcloud:Configuration')
+          conf.xpath('./vcloud:FenceMode').text.must_equal 'bridged'
+          conf.xpath('./vcloud:ParentNetwork/@href').text.must_equal '/network/parent-network-id'
```

Comment:
Just making sure: when the real request will be made, this href will also contain complete VCD URL, not just the path part? [Here](https://github.com/xlab-si/fog-vcloud-director/pull/3/files?diff=unified#diff-17521a58b31b5106b914ecef49d50e89R44) you are pushing `endpoint` to the `options` giving the request a chance to compose entire `href` [here](https://github.com/xlab-si/fog-vcloud-director/pull/3/files?diff=unified#diff-a38898daacbba99835e64d468b9e4cd8R15)

Reply:
Correct. Thanks for pointing this out, I've updated the test to use `ENDPOINT` text instead of just empty string so that we now expect string `'ENDPOINT/network/parent-network-id'`.